src/api/CoinBase.ts

```diff
@@ -0,0 +1,26 @@
+import { CoinBaseResponse, SourceApi } from '../types/RelayPricer';
+import fetch, { Response } from 'node-fetch';
+
+export default class CoinBase implements SourceApi {
+    // we should get this address from config file or enviroments
+    private readonly URL: string = 'https://api.coinbase.com/v2/exchange-rates';
+
+    async query(input: string, output: string): Promise<number> {
+        const upperCaseOutput = output.toUpperCase();
+        const response: Response = await fetch(
+            `${this.URL}?currency=${input}`,
+            {
+                method: 'get'
+            }
+        );
+
+        if (response.status === 200) {
+            const result: CoinBaseResponse = await response.json();
+            if (!(upperCaseOutput in result.data.rates)) {
+                throw new Error('Output not available');
+            }
+            return Number(result.data.rates[upperCaseOutput]);
```

Comment:
just a knitpicking optimisation, but I think it would be better as:
```suggestion
            const rate = rates.find(rate => rate.toUpperCase() === output.toUpperCase()); // just to make sure the casing is the same, apply to both
            if (!rate) {
                throw new Error(`Exchange rate for currency pair ${input}/${output} is not available`);
            }
            return Number(rate);
```

Reply:
rates it not an array. its an object. Ill reuse the error message.

Reply:
updated